nestedscrollwebview/src/main/java/com/telefonica/nestedscrollwebview/helper/InternalScrollDetector.kt

```diff
@@ -0,0 +1,80 @@
+package com.telefonica.nestedscrollwebview.helper
+
+import android.view.MotionEvent
+
+class InternalScrollDetector {
+
+    private var isScrolling: Boolean = false
+    private var pageScrollChangedWhileScrolling: Boolean = false
+    private var initialX: Float? = null
+    private var initialY: Float? = null
+    private var activePointerId: Int = INVALID_POINTER
+
+    private var isEnabled = false
+
+    fun setEnabled(enabled: Boolean) {
+        if (isEnabled != enabled) {
+            reset()
+        }
+        isEnabled = enabled
+    }
+
+    fun onTouchEvent(event: MotionEvent): Boolean {
+        if (!isEnabled) {
+            return false
+        }
+
+        return when (event.action) {
+            MotionEvent.ACTION_DOWN -> {
+                reset()
+                initialX = event.x
+                initialY = event.y
+                activePointerId = event.getPointerId(0)
+                false
+            }
+            MotionEvent.ACTION_MOVE -> {
+                val activePointerIndex = event.findPointerIndex(activePointerId)
+                if (activePointerIndex == -1 || initialX == null || initialY == null) {
+                    false
+                } else {
+                    val y = event.getY(activePointerIndex)
+                    val x = event.getX(activePointerIndex)
+                    val deltaY: Float = initialY!! - y
+                    val deltaX: Float = initialX!! - x
```

Comment:
It shouldn't because all class methods should be always executed from main thread (All are related with touch events processing or scrolls performed that will be sequentially invoked)